test cases/common/183 as-needed/main.cpp

```diff
@@ -0,0 +1,7 @@
+#include <cstdlib>
+
+#include "libA.h"
+
+int main() {
+  return (meson_test_as_needed::linked == false ? EXIT_SUCCESS : EXIT_FAILURE);
```

Comment:
`!meson_test_as_needed::linked` is usually cleaner than comparing to another boolean